ir/ir.def

```diff
@@ -178,6 +178,25 @@ class KeyElement {
     Annotations     annotations;
     Expression      expression;
     PathExpression  matchType;
+
+    const IR::Node *transform_visit(Transform &v) {
```

Comment:
I expect most backends will need at least one pass that uses this, in a way that makes doing it in a common midend pass difficult.  Ideally this would be handled automatically by the ir-generator, but I don't have a good way of getting it to do this.